phoenicis-apps/src/main/java/org/phoenicis/apps/GitApplicationsSource.java

```diff
@@ -18,45 +18,109 @@
 
 package org.phoenicis.apps;
 
-import org.phoenicis.apps.dto.CategoryDTO;
-
 import java.io.File;
 import java.io.IOException;
-import java.nio.file.Files;
-import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.errors.RepositoryNotFoundException;
+import org.phoenicis.apps.dto.CategoryDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 class GitApplicationsSource implements ApplicationsSource {
-    private final String gitRepositoryURL;
-    private final LocalApplicationsSource.Factory localAppsSourceFactory;
-    private List<CategoryDTO> cache;
-
-    GitApplicationsSource(String gitRepositoryURL, LocalApplicationsSource.Factory localAppsSourceFactory) {
-        this.gitRepositoryURL = gitRepositoryURL;
-        this.localAppsSourceFactory = localAppsSourceFactory;
-    }
-
-    @Override
-    public synchronized List<CategoryDTO> fetchInstallableApplications() {
-        if (cache != null) {
-            return cache;
-        }
-
-        try {
-            final File gitTmp = Files.createTempDirectory("git").toFile();
-            gitTmp.deleteOnExit();
-            new ProcessBuilder(Arrays.asList("git", "clone", gitRepositoryURL, gitTmp.getAbsolutePath()))
-                    .inheritIO()
-                    .start()
-                    .waitFor();
-
-            cache = localAppsSourceFactory.createInstance(gitTmp.getAbsolutePath()).fetchInstallableApplications();
-            return cache;
-        } catch (IOException | InterruptedException e) {
-            return Collections.emptyList();
-        }
-    }
+	private final static Logger LOGGER = LoggerFactory.getLogger(GitApplicationsSource.class);
+
+	private final String gitRepositoryURL;
+	private final LocalApplicationsSource.Factory localAppsSourceFactory;
+	private List<CategoryDTO> cache;
+
+	private static final String cacheDirectoryPath = System.getProperty("java.io.tmpdir");
+
+	GitApplicationsSource(String gitRepositoryURL, LocalApplicationsSource.Factory localAppsSourceFactory) {
+		this.gitRepositoryURL = gitRepositoryURL;
+		this.localAppsSourceFactory = localAppsSourceFactory;
+	}
+
+	@Override
+	public synchronized List<CategoryDTO> fetchInstallableApplications() {
+		if (cache != null) {
+			return cache;
+		}
+
+		File gitRepositoryLocation = new File(
+				GitApplicationsSource.cacheDirectoryPath + "/git" + this.gitRepositoryURL.hashCode());
+
+		LOGGER.info(String.format("Git-repository '%s' will be saved in '%s'", this.gitRepositoryURL,
+				gitRepositoryLocation.getAbsolutePath()));
+
+		/*
```

Comment:
This is not meant as a comment for the reader of the code, but as a debug message, in case something went wrong.